src/middleware.ts

```diff
@@ -401,37 +411,37 @@ const toThenable =
   }
 
 export const persist =
-  <S extends State>(config: StateCreator<S>, options: PersistOptions<S>) =>
-  (set: SetState<S>, get: GetState<S>, api: StoreApi<S>): S => {
-    const {
-      name,
-      getStorage = () => localStorage,
-      serialize = JSON.stringify as (state: StorageValue<S>) => string,
-      deserialize = JSON.parse as (str: string) => StorageValue<Partial<S>>,
-      blacklist,
-      whitelist,
-      partialize = (state: S) => state,
-      onRehydrateStorage,
-      version = 0,
-      migrate,
-      merge = (persistedState: any, currentState: S) => ({
+  <S extends State>(config: StateCreator<S>, _options: PersistOptions<S>) =>
+  (set: SetState<S>, get: GetState<S>, api: PersistApi<S>): S => {
+    let options = {
```

Comment:
```suggestion
    let options: PersistOptions<S> = {
```
just an extra type checking.

Reply:
`PersistOptions` keeps optional fields as `<T> | undefined` even if they have a default value. So we need to do `storage = options?.getStorage()` for instance because typescript think `getStorage` is possibly `undefined` even if it has a default value.
Making this change would require to add unneeded conditions just to satisfy typescript.

Reply:
I didn't notice that. Then, my suggestion is void.